### DISCUSSION TITLE
How do I save an image to a device's gallery

### DISCUSSION THREAD

**Discussion by hunterInt** [ionic-team/capacitor#4487] [Q&A]
I'm trying to save an image (or video) to the gallery so that the image can be accessed from other apps that use the devices gallery. How can I do this? This is what I've tried: 
```
await Camera.checkPermissions();
await Camera.requestPermissions();
await Filesystem.checkPermissions();
await Filesystem.requestPermissions();

const savedFile = await Filesystem.writeFile({
	path: fileName,// foo.jpg
	data: base64Data,
	directory: Directory.Data
}).catch(console.error);
```
I'm aware that I can save an image to the gallery using the [Camera plugin](https://capacitorjs.com/docs/v3/apis/camera#example), but I don't want to take a picture I already have a picture that I'd like to save. `Directory.Data` just seems like the local private directory and anything saved there is not accessed from other apps, which is not what I would like.

**Comment by techycereal** [comment]
I know its been three years but in case you haven't figured it out here's the answer:


Start by installing the Capacitor Community Media plugin: npm install @capacitor-community/media

In your code, import the Media module: import { Media } from "@capacitor-community/media";

Here's how you can ensure an album exists and save an image to it within your App.jsx:

const ensureDemoAlbum = async () => {
        const { albums } = await Media.getAlbums();
        let demoAlbum = undefined;
        if (Capacitor.getPlatform() === "android") {
            const albumsPath = (await Media.getAlbumsPath()).path
            demoAlbum = albums.find(a => a.name === "Media Demo Album" && a.identifier.startsWith(albumsPath));
            console.log(demoAlbum);
        } else {
            demoAlbum = albums.find(a => a.name === "Media Demo Album");
        }
    
        if (!demoAlbum) {
            console.log(`Demo album does not exist; create it first using the "Create Demo Album" button above.`);
            throw new Error("Demo album does not exist");
        }
    
        return demoAlbum.identifier;
    }

const savedFile = await Filesystem.writeFile({
            path: fileName,
            data: base64Data,
            directory: Directory.External // Save to app's private data directory
          });
          
          let opts = { path: savedFile.uri,albumIdentifier: await ensureDemoAlbum(), fileName: fileName };
          await Media.savePhoto(opts);



Before saving images, make sure the album exists. Here's how you can create a "Media Demo Album" using a button in your CreateAlbum.jsx:

import { Media } from '@capacitor-community/media';
const createDemoAlbum = async () => {
        setStatus("");
        const { albums } = await Media.getAlbums();
        const demoAlbum = albums.find(a => a.name === "Media Demo Album");
        if (demoAlbum) {
            setStatus("Demo album already exists!");
            return;
        }

        await Media.createAlbum({ name: "Media Demo Album" });
        setStatus("Created demo album");
        console
    }
    
    <button onClick={createDemoAlbum}>Create Demo Album</button>
    
    
    
Integrate the CreateAlbum component within your App.jsx to provide a user interface for album creation.
